### ISSUE TITLE
Rewarded ads - iPhone - No callbacks when the audio is muted

### ISSUE THREAD

**Issue by edscodeblenders** [googleads/googleads-mobile-unity#559]
Hi there,

When we show a rewarded ad and the user mutes the audio of the video, the video is watched to the end, but the callback to get the reward is not received. This happens ALWAYS when you mute the audio, and works fine when the audio is on.

On Android this works fine in both cases, we receive the callback even when the video is played without audio.

Is this a known issue?

Regards.

**Comment by rampara** [comment]
@edscodeblenders I'm having trouble reproducing this issue. Can you provide step by step instructions on how you were able to reproduce this behavior?

**Comment by edscodeblenders** [comment]
Hi @rampara,

What we do is very simple, it happens in iPhone only:
1) We request a rewarded video ad.
2) When it's ready we show it.
3) When the video starts playing, we press the "mute" button, to mute the sound.
4) We watch the video to the end.
5) When the video is complete, we close the ad, but the "OnAdRewarded" callback is not received.

On the other hand, if we do the same except point 3, we receive the callback without any issue.

We are reproducing this in our game. I'm trying to reproduce the same with the HelloWorld sample, but we can't get rewarded ads from Admob using the test ad unit "ca-app-pub-3940256099942544/1712485313". We get this error:
2017-11-15 17:35:27.509028-0300 ProductName[4140:1300495] <Google> Cannot find an ad network adapter with the name(s): com.google.DummyAdapter. Remember to link all required ad network adapters and SDKs, and set -ObjC in the 'Other Linker Flags' setting of your build target.
HandleRewardBasedVideoFailedToLoad event received with message: Failed to receive ad with error: No ad returned from any ad server.
GoogleMobileAds.iOS.RewardBasedVideoAdClient:RewardBasedVideoAdDidReceiveAdCallback(IntPtr)
System.Array:SetGenericValueImpl(Int32, T&)

If you help us with the error above maybe we can reproduce the issue with the test app provided by you.

Thanks.

**Comment by rampara** [comment]
@edscodeblenders the error you're running into is caused when there is no eligible to be served for your ad request. There was an issue with our test ads that since been resolved. Can you try reproducing with the HelloWorld sample app again?

**Comment by edscodeblenders** [comment]
Hi @rampara, thanks for your feedback and help. We couldn't reproduce the issue with the test app provided with the SDK, so we started debugging our game, and we found that the issue was related to our implementation. We were requesting to load a second rewarded video ad before the first one was closed, so even if the ad was working, and we were watching the video to the end, the rewarded / closed callbacks were not being triggered, because they were for a previous ad. After ensuring that the ad is completely closed before requesting a new ad, the issue is gone. Thanks!
